tests/e2e/fixtures/storage_fsx_dependencies.py

```diff
@@ -51,179 +42,63 @@ def get_fsx_mount_name(fsx_client, file_system_id):
     return response["FileSystems"][0]["LustreConfiguration"]["MountName"]
 
 
-@pytest.fixture(scope="class")
-def install_fsx_csi_driver(metadata, region, request, cluster, kustomize):
-    fsx_driver = {}
-    FSx_DRIVER_VERSION = "v0.7.1"
-    FSx_CSI_DRIVER = f"github.com/kubernetes-sigs/aws-fsx-csi-driver/deploy/kubernetes/overlays/stable/?ref=tags/{FSx_DRIVER_VERSION}"
+def get_file_system_id_from_name(fsx_client, file_system_name):
+    def name_matches(filesystem):
+        return filesystem["Name"] == file_system_name
 
-    def on_create():
-        kubectl_apply_kustomize(FSx_CSI_DRIVER)
-        fsx_driver["driver_version"] = FSx_DRIVER_VERSION
+    file_systems = fsx_client.describe_file_systems()["FileSystems"]
 
-    def on_delete():
-        kubectl_delete_kustomize(FSx_CSI_DRIVER)
+    file_system = next(filter(name_matches, file_systems))
```

Comment:
Can you explain this line to me? I see that we are filtering for a name_match which will return an iterable but why are we calling next here?